custom_components/lg_tv_serial/__init__.py

```
@@ -54,19 +55,47 @@ async def async_send_raw(call):
+                            try:
+                                parsed_value = int(str(value).strip(), 0)  # 0 = auto base
+                                if 0 <= parsed_value <= 255:
+                                    data.append(parsed_value)
+                                else:
+                                    raise ValueError
+                            except (TypeError, ValueError):
+                                raise ServiceValidationError(
+                                    translation_domain=DOMAIN,
+                                    translation_key="invalid_data_value",
+                                    translation_placeholders={"data_byte": attr, "wrong_value": value},
+                                )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)